src/invert/parderiv/impls/cyclic/cyclic.cxx

```diff
@@ -68,8 +65,7 @@ const Field3D InvertParCR::solve(const Field3D &f) {
   Field3D alignedField = toFieldAligned(f, RGN_NOX);
 
   // Create cyclic reduction object
-  CyclicReduce<dcomplex> *cr = 
-    new CyclicReduce<dcomplex>();
+  auto* cr = new CyclicReduce<dcomplex>();
```

Comment:
`unique_ptr`?

Also, would it make sense to have `cr` as a member of `InvertParCR` so it doesn't have to be allocated/deleted each time `solve()` is called?

Reply:
It looks like a lot of what `InvertParCR` does could be cached, so I think this is ok just being `unique_ptr`